api/stacks/deploy.go

```diff
@@ -57,20 +57,13 @@ func RedeployWhenChanged(stackID portainer.StackID, deployer StackDeployer, data
 	if strings.EqualFold(newHash, string(stack.GitConfig.ConfigHash)) {
 		return nil
 	}
-
-	cloneParams := &cloneRepositoryParameters{
-		url:   stack.GitConfig.URL,
-		ref:   stack.GitConfig.ReferenceName,
-		toDir: stack.ProjectPath,
-	}
-	if stack.GitConfig.Authentication != nil {
-		cloneParams.auth = &gitAuth{
-			username: username,
-			password: password,
-		}
-	}
-
-	if err := cloneGitRepository(gitService, cloneParams); err != nil {
+	if err := gitService.CloneRepository(
+		stack.ProjectPath,
+		stack.GitConfig.URL,
+		stack.GitConfig.ReferenceName,
+		username,
+		password,
+	); err != nil {
 		return errors.WithMessagef(err, "failed to do a fresh clone of the stack %v", stack.ID)
 	}
```

Comment:
I'm very frustrated that there is no attempt to log some git info here, and add it to some form of CD log - its been very hard to work out what specific git SHA was deployed, and I hit some "deploy too soon after git push" pain that I couldn't debug.